### ISSUE TITLE
Metric base class does not check for extra kwargs

### ISSUE THREAD

**Issue by EPronovost** [Lightning-AI/metrics#1426]
## 🐛 Bug

The `Metric` base class does not check for extra kwargs [here](https://github.com/Lightning-AI/metrics/blob/bc057d495703aff5c4d8d671c54060e08a677142/src/torchmetrics/metric.py#L94-L120).
Users can currently do something like `torchmetrics.MeanMetric(foo=True)` without issue.

### To Reproduce

```
import torchmetrics
torchmetrics.MeanMetric(your_favorite_word="lightning")
```

### Expected behavior

Raise an exception if invalid kwargs are passed

### Environment

Latest TorchMetrics version. No other dependencies matter.

### Additional context

[Slack thread](https://pytorch-lightning.slack.com/archives/C010R4A7189/p1672859410056429)
